### PR TITLE
chore: use cmake exported targets

### PR DESCRIPTION
**Pull request by Amadrath**
Tested with generic-chimeratk-server

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -105,34 +99,35 @@ include_directories(${PROJECT_BINARY_DIR}/epics-base/include)
...
 endforeach()
-target_link_libraries(${PROJECT_NAME} ${PROJECT_LIB_DIR}/libChimeraTK-ControlSystemAdapter-EPICS.a)
-target_link_libraries(${PROJECT_NAME} ${readline_LIBRARY})
+target_link_libraries(${PROJECT_NAME} PRIVATE ChimeraTK::ChimeraTK-ControlSystemAdapter)
+target_link_libraries(${PROJECT_NAME} PRIVATE ${PROJECT_LIB_DIR}/libChimeraTK-ControlSystemAdapter-EPICS.a)
+target_link_libraries(${PROJECT_NAME} PRIVATE readline)
 
 if(EPICS_VER STREQUAL "epics7")
-  target_link_libraries(${PROJECT_NAME} ${libYajl_LIBRARY})
+  #NOTICE Due to https://github.com/lloyd/yajl/issues/209 the correct linker flag is -lyajl_s.
```
Comment (mhier) [inline CMakeLists.txt:112]: readability: add space after # character

File: CMakeLists.txt [hunk 2]
```diff
@@ -105,34 +99,35 @@ include_directories(${PROJECT_BINARY_DIR}/epics-base/include)
...
+  #TODO If this issue is resolved, the library might have to be changed from yail_s to yajl!
+  target_link_libraries(${PROJECT_NAME} PUBLIC yajl_s)
 endif()
 
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${${PROJECT_NAME}_FULL_LIBRARY_VERSION} SOVERSION ${${PROJECT_NAME}_SOVERSION})
-set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS "${ChimeraTK-ControlSystemAdapter_LINKER_FLAGS} -Wl,--no-as-needed")
+#set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS "${ChimeraTK-ControlSystemAdapter_LINKER_FLAGS} -Wl,--no-as-needed")
```
Comment (mhier) [inline CMakeLists.txt:118]: please remove commented code lines

**Review by mhier — APPROVED** [review]
Two minor things, otherwise please go ahead with the merge (use the edited PR title as commit message e.g. by squashing)

File: CMakeLists.txt (hunk 1 above)
Comment (Amadrath) [reply CMakeLists.txt]: Done

**Pull request merged**